docs/cow-protocol/tutorials/cow-swap/swap.mdx

```diff
@@ -10,6 +10,28 @@ import ReceiverAndOrderWarning from '../../../partials/_receiver.mdx';
 
 Swapping is the bread and butter of CoW Protocol! Let's swap some tokens!
 
+## Connect Wallet
+Go to [swap.cow.fi](https://swap.cow.fi/#/swap?utm_source=docs.cow.fi&utm_content=swap-tutorial) and connect your web3 wallet. 
+
+:::note
+
+If you don't have a web3 wallet yet, you can easily get one!
+- Use a browser extension wallet such as [Rabby](https://rabby.io/) or [MetaMask](https://metamask.io/download.html).
+- A hardware wallet such as [Ledger](https://www.ledger.com/) or [Trezor](https://trezor.io/).
+- A Smart Contract Wallet such as [Safe](https://app.safe.global)
+
+Please do your own research on the trade offs of each wallet to select the one that works best for you.
```

Comment:
```suggestion
Please do your own research on the advantages and disadvantages of each wallet to select the one that works best for you.
```

nit: above suggestion to get away from "trade off" which can be contentious about whether it is hyphenated or not :laughing: